src/wallet/fees.cpp

```diff
@@ -66,7 +69,7 @@ CAmount GetMinimumFee(unsigned int nTxBytes, const CCoinControl& coin_control, c
     CAmount required_fee = GetRequiredFee(nTxBytes);
     if (required_fee > fee_needed) {
         fee_needed = required_fee;
-        if (feeCalc) feeCalc->reason = FeeReason::REQUIRED;
+        if (feeCalc && feeCalc->reason != FeeReason::FALLBACK) feeCalc->reason = FeeReason::REQUIRED;
```

Comment:
How is this not dead code?

Reply:
Why would that be dead code?

Reply:
Excuse my brevity, but I fail to see why fallbackfee is special cased here.

Let's assume the user sets paytxfee to one satoshi per byte, but the required fee is higher. It will correctly report that the required fee rate was used.
Let's assume the user sets fallbackfee to something very low and the required fee is higher. It will incorrectly report in the debug.log that the fallbackfee was used, no? Since in this case `g_wallet_allow_fallback_fee` will be true, it won't incorrectly hit the check [in wallet.cpp](https://github.com/bitcoin/bitcoin/pull/11882/files#diff-b2bb174788c7409b671c46ccc86034bdR2875).

So if there is a reason to add this piece of code, please explain and/or add a comment.

Reply:
You right. It's no longer in use.
The question here is, that we return a single flag where a bitset would be required.
Reason can be FeeReason::REQUIRED but also (AND) FeeReason::FALLBACK.

This code part was required to enable RBF125 in case a fallback fee has been used regardless of the mintxrelayfee.
But no longer relevant, will remove.

Reply:
Fixed.